components/menu/menu.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode, useState, MouseEvent} from 'react';
+import {Menu, MenuList} from '@mui/material';
+
+import Tooltip from 'components/tooltip';
+import CompassDesignProvider from 'components/compass_design_provider';
+import OverlayTrigger from 'components/overlay_trigger';
+
+import type {PropsFromRedux} from './index';
+
+const OVERLAY_TIME_DELAY = 500;
+
+interface Props extends PropsFromRedux {
+
+    // Anchor button props
+    anchorId?: string;
+    anchorNode?: ReactNode;
+    anchorClassName?: string;
+    anchorAriaLabel?: string;
+
+    // Menu props
+    menuId?: string;
+    menuAriaLabel?: string;
+
+    // Tooltip props
+    tooltipPlacement?: 'top' | 'bottom' | 'left' | 'right';
+    tooltipId?: string;
+    tooltipText?: string;
+    tooltipClassName?: string;
+
+    children: ReactNode;
+}
+
+export function MenuComponent(props: Props) {
+    const [anchorElement, setAnchorElement] = useState<null | HTMLElement>(null);
+    const isMenuOpen = Boolean(anchorElement);
+
+    const handleMenuOpen = (event: MouseEvent<HTMLButtonElement>) => {
+        event.preventDefault();
+        setAnchorElement(event.currentTarget);
+    };
+
+    const handleMenuClose = (event: MouseEvent<HTMLDivElement | HTMLUListElement>) => {
+        event.preventDefault();
+        setAnchorElement(null);
+    };
+
+    return (
+        <CompassDesignProvider theme={props.theme}>
```

Comment:
Do we need to target individual menus with this? I would've thought that we'd wrap the whole app in this similar to how we do the react-intl context

Reply:
i have tried that but somehow the context of provider placed in root is not reaching way down in the component. 🤔

Reply:
I wonder if it's similar to the old `OverlayTrigger` that doesn't pass context down properly any more. Is it just submenus that it doesn't properly apply to?

Reply:
Its strange that it does not pass it down. I wonder what could be hindering it from doing so. Would you mind trying to add this to the routing level? The last time I checked that was the place where I tried to add it and it worked.